### ISSUE TITLE
Unreliable Major Version Detection

### ISSUE THREAD

**Issue by davidalger** [netz98/n98-magerun#465]
When the tool initializes itself, it makes an attempt to detect the major version number of the system. If the major version detected is v2 then it spits out a notice with instructions on using on a new version of this tool. The check is currently performed by counting the files present in the ./app/ directory with names matching Mage.php, bootstrap.php and autoloader.php. If it counts 2 files it detects v2, otherwise defaults to v1. This causes a bit of a problem for some sites where either one of these other files is present in an installation of Magento v1, preventing the tool from working. We've used a bootstrap.php file on sites with a dozen or more entry points. A workaround allowing us to use the tool is to create an empty file named autoloader.php as well to avoid being detected as v2.

Now that Magento 2 is nearing stability, it seems having a more proper version check in place is appropriate. I'm open to suggestions on what we should check and **may** be able to PR it myself depending on the level of effort required.

**Comment by AydinHassan** [comment]
Would it not work to just parse `Mage.php` and check the static version variables? Or just assume Magento 1 if no `app/Mage.php` is present as that doesn't appear to be in Magento 2?

**Comment by cmuench** [comment]
@AydinHassan you are right. The current code is very old. The first public versions of Magento 2 contained a Mage.php file. In the actual version the "god class" is not available anymore. So we can simply check the single file.

**Comment by davidalger** [comment]
@cmuench Good point about it having still existed in Magento 2 at one point. I had been wondering why the original check wasn't written that way. :)

@AydinHassan I tried that, but the autoloader does not load the file from this point in the initialization process.
I just changed the logic a bit and it checks out ok on my instances. I have not checked the unit tests, so we'll see what Travis comes back with. Left the search as is but checking to see if Mage.php is in the list of files found and if not assuming Magento is v2.x.
